### ISSUE TITLE
event-handlers don't appear to be returning Subscription instance

### ISSUE THREAD

**Issue by stharvey** [transistorsoft/react-native-background-geolocation#1447]
## Your Environment
* Plugin version: 4.4.4
* Platform: Android
* OS version: 12
* Device manufacturer / model: Google / Pixel 6 Pro
* React Native version (`react-native -v`): 0.63.4
* Plugin config
```javascript <!-- Syntax-highlighting:  paste your code below -->
useEffect(() => {
    const onLocation = BackgroundGeolocation.onLocation((location) => {
      onLocationDo(location);
    });
    console.log(onLocation); //undefined

    const onMotionChange = BackgroundGeolocation.onMotionChange((event) => {
      onMotionChangeDo(event);
    });

    const onActivityChange = BackgroundGeolocation.onActivityChange((event) => {
      onActivityChangeDo(event);
    });

    const onProviderChange = BackgroundGeolocation.onProviderChange((event) => {
      onProviderChangeDo(event);
    });


    return () => {
      console.log(onLocation); //undefined
      onLocation.remove();
      onMotionChange.remove();
      onActivityChange.remove();
      onProviderChange.remove();
    };
  }, []);
```

## Expected Behavior
I expect subscription instances to be created and removed during useEffect

## Actual Behavior
The subscription instances appear to be returned undefined and when I perform a live reload I receive:
```
TypeError: Cannot read property 'remove' of undefined

This error is located at:
    in GeoService (at App.js:225)
    in App (created by ExpoRoot)
    in ExpoRoot (at renderApplication.js:45)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:106)
    in RCTView (at View.js:34)
    in View (at AppContainer.js:132)
    in AppContainer (at renderApplication.js:39), js engine: hermes
```

## Context
I have just updated the plugin from 4.1.1 to 4.4.4 however I can't figure out what I am doing wrong with the new subscription interface.  It appears that the subscriptions are not being returned - though it's likely that I am not implementing it correctly - I just can't see what I am doing wrong - My project is an Expo bare workflow without TypeScript.

**Comment by stharvey** [comment]
Please ignore this issue. I was trying to set the subscriptions before calling BackgroundGeolocation.ready - at least I think that's the reason it wasn't working! Working great now.

**Comment by stharvey** [comment]
Hello again,

I'm sorry for reopening this but I was mistaken when I thought I had resolved the issue.

It must be something to do with my configuration or plugin versions but I can't find the issue.

Within my app I have created a new component using the same code as in the docs example:

`import React from "react";
import { Switch, Text, View } from "react-native";

import BackgroundGeolocation, {
  Location,
  Subscription,
} from "react-native-background-geolocation";

const TestGeo = () => {
  const [enabled, setEnabled] = React.useState(false);
  const [location, setLocation] = React.useState("");

  React.useEffect(() => {
    /// 1.  Subscribe to events.
    const onLocation: Subscription = BackgroundGeolocation.onLocation(
      (location) => {
        console.log("[onLocation]", location);
        setLocation(JSON.stringify(location, null, 2));
      }
    );

    console.log(onLocation); //undefined


    const onMotionChange: Subscription = BackgroundGeolocation.onMotionChange(
      (event) => {
        console.log("[onMotionChange]", event);
      }
    );

    const onActivityChange: Subscription =
      BackgroundGeolocation.onActivityChange((event) => {
        console.log("[onMotionChange]", event);
      });

    const onProviderChange: Subscription =
      BackgroundGeolocation.onProviderChange((event) => {
        console.log("[onProviderChange]", event);
      });

    /// 2. ready the plugin.
    BackgroundGeolocation.ready({
      // Geolocation Config
      desiredAccuracy: BackgroundGeolocation.DESIRED_ACCURACY_HIGH,
      distanceFilter: 10,
      // Activity Recognition
      stopTimeout: 5,
      // Application config
      debug: true, // <-- enable this hear sounds for background-geolocation life-cycle.
      logLevel: BackgroundGeolocation.LOG_LEVEL_VERBOSE,
      stopOnTerminate: false, // <-- Allow the background-service to continue tracking when user closes the app.
      startOnBoot: true, // <-- Auto start tracking when device is powered-up.
      // HTTP / SQLite config
      url: "http://yourserver.com/locations",
      batchSync: false, // <-- [Default: false] Set true to sync locations to server in a single HTTP request.
      autoSync: true, // <-- [Default: true] Set true to sync each location to server as it arrives.
      headers: {
        // <-- Optional HTTP headers
        "X-FOO": "bar",
      },
      params: {
        // <-- Optional HTTP params
        auth_token: "maybe_your_server_authenticates_via_token_YES?",
      },
    }).then((state) => {
      setEnabled(state.enabled);
      console.log(
        "- BackgroundGeolocation is configured and ready: ",
        state.enabled
      );
    });

    return () => {
      // Remove BackgroundGeolocation event-subscribers when the View is removed or refreshed
      // during development live-reload.  Without this, event-listeners will accumulate with
      // each refresh during live-reload.
      onLocation.remove();
      onMotionChange.remove();
      onActivityChange.remove();
      onProviderChange.remove();
    };
  }, []);

  /// 3. start / stop BackgroundGeolocation
  React.useEffect(() => {
    if (enabled) {
      BackgroundGeolocation.start();
    } else {
      BackgroundGeolocation.stop();
      setLocation("");
    }
  }, [enabled]);

  return (
    <View style={{ alignItems: "center" }}>
      <Text>Click to enable BackgroundGeolocation</Text>
      <Switch value={enabled} onValueChange={setEnabled} />
      <Text style={{ fontFamily: "monospace", fontSize: 12 }}>{location}</Text>
    </View>
  );
};

export default TestGeo;
`

If I `console.log(onLocation);` I always get `undefined`.

I am using:
```
   "react-native-background-fetch": "^4.0.5",
    "react-native-background-geolocation": "^4.4.4",
```

The event handlers do appear to be working but they are not returning a subscription instance.

Can you point me in the right direction of where I should be looking for a solution? Thanks

**Comment by christocracy** [comment]
Create yourself a fresh new Hello World app and see if the problem persists.

**Comment by stharvey** [comment]
Thanks Chris,

Yes, creating a new fresh app (with a more recent version of RN) it was working fine.  Back in the original app I placed the code example as the only component in the app and I was still getting the same issue.

I was using react-native: 0.63.4

Anyway, I initiated a new app with react-native: 0.64.3 and have moved all my code across and everything is working great now. I don't know if it was related to RN version or if it was related to another package or dependency (I needed to update a couple of dozen packages to get the app running correctly).
